Added support for CIT option

shailesh-kalamkar opened the pull request:
@arpit1094 _A Please review.

shailesh-kalamkar committed:
Support for CIT option

1.Added base class for here maps APIs.
2.Added changes to support CIT option and change API domain name based on it.
3. Overrided here.rb lookup file to support CIT option and change API domain name based on it.

arpit1094 commented:
@shailesh-kalamkar , I was going through geocoder gem code. Will it make more sense to create a setting `use_cit` like https://github.com/alexreisner/geocoder/blob/master/lib/geocoder/configuration.rb ?This will also save us the hassle of passing `cit` option everywhere.

shailesh-kalamkar commented:
@arpit1094 We will need to dynamically switch the `cit` option based on the environment, so don't know how configuration will fit for it.

arpit1094 commented:
@shailesh-kalamkar , that can be controlled by environment files like `app_id` and `app_code`

Commit:
wip

Commit:
Uses cit from geocoder.rb settings instead of explicit option
Refactores here lookup code to avoid overwriting whole class
Refactores domain modification code
Refactors here_base.rb code to DefaultLookupMethods concern
Uses :here handle by default to avoid defining individual configuration for every lookup

arpit1094 commented:
@shailesh-kalamkar _A, please review.

shailesh-kalamkar commented on lib/geocoder/lookups/default_lookup_methods.rb:
```diff
@@ -0,0 +1,43 @@
+module Geocoder::Lookup
+  module DefaultLookupMethods
```
@arpit1094 This file is `here` maps related. The module name/namespace should suggest that.

shailesh-kalamkar reviewed (commented):
@arpit1094 Added one comment. Please check.

Commit:
-Renames LookupDefaultMethods
-Bumps version to 0.2.0

arpit1094 commented:
@shailesh-kalamkar , done with the changes. Please review.

shailesh-kalamkar commented:
@arpit1094  Looks good to me.
